src/components/CheckEditor/CheckSettings.tsx

```diff
@@ -41,29 +48,52 @@ export default class CheckSettings extends PureComponent<Props, State> {
     this.setState({ settings: settings }, this.onUpdate);
   };
 
-  onSettingsChange = (settings: Settings) => {
-    this.setState({ settings: settings }, this.onUpdate);
+  onSettingsChange = (settings: Settings, labels: Label[]) => {
+    this.setState({ settings, labels }, this.onUpdate);
   };
 
   render() {
-    const { settings } = this.state;
+    const { settings, labels } = this.state;
     if (!settings) {
       return <div>Loading....</div>;
     }
     const { isEditor } = this.props;
 
     switch (this.props.typeOfCheck) {
       case CheckType.PING: {
-        return <PingSettingsForm settings={settings} onUpdate={this.onSettingsChange} isEditor={isEditor} />;
+        return (
+          <PingSettingsForm labels={labels} settings={settings} onUpdate={this.onSettingsChange} isEditor={isEditor} />
+        );
       }
       case CheckType.HTTP: {
-        return <HttpSettingsForm settings={settings} onUpdate={this.onSettingsChange} isEditor={isEditor} />;
+        return (
+          <HttpSettingsForm
+            labels={labels ?? []}
+            settings={settings}
+            onUpdate={this.onSettingsChange}
+            isEditor={isEditor}
+          />
+        );
       }
       case CheckType.DNS: {
-        return <DnsSettingsForm settings={settings} onUpdate={this.onSettingsChange} isEditor={isEditor} />;
+        return (
+          <DnsSettingsForm
+            labels={labels ?? []}
+            settings={settings}
+            onUpdate={this.onSettingsChange}
+            isEditor={isEditor}
+          />
+        );
       }
       case CheckType.TCP: {
-        return <TcpSettingsForm settings={settings} onUpdate={this.onSettingsChange} isEditor={isEditor} />;
+        return (
+          <TcpSettingsForm
+            settings={settings}
+            labels={labels ?? []}
+            onUpdate={this.onSettingsChange}
+            isEditor={isEditor}
+          />
+        );
```

Comment:
Not a blocker or anything, but interesting idea about TS:
There's a way to make a `switch` statement or a series of `if`s exhaustive, causing compiler to throw an error if not all  possible values of an enum are covered:
https://dev.to/babak/exhaustive-type-checking-with-typescript-4l3f

I like to use it, to make changing enum values safer in the future